### PR TITLE
feat: expose prefabs for newly added targets

### PR DESCRIPTION
**Pull request by WoLewicki**
## Summary:

In recent commits, some new targets have been added, and they are not exposed as prefabs, yet are used in e.g. `TextLayoutManager`. They are needed then for `react-native-live-markdown`: https://github.com/Expensify/react-native-live-markdown/pull/428/commits/c1611cd98ed5009fd66c871b9999b55941086af0

## Changelog:

[ANDROID] [ADDED] - expose prefabs for newly added targets

## Test Plan:

It cannot be tested inside the repo, but try to build the `example` app with new arch enabled on `Android` in the `react-native-live-markdown` repo to see that those are needed.

### CODE REVIEW

File: packages/react-native/ReactAndroid/cmake-utils/ReactNative-application.cmake [hunk 1]
```diff
@@ -87,6 +87,11 @@ add_library(react_nativemodule_core ALIAS ReactAndroid::react_nativemodule_core)
 add_library(react_render_imagemanager ALIAS ReactAndroid::react_render_imagemanager)
 add_library(rrc_image ALIAS ReactAndroid::rrc_image)
 add_library(rrc_legacyviewmanagerinterop ALIAS ReactAndroid::rrc_legacyviewmanagerinterop)
+add_library(reactnativejni ALIAS ReactAndroid::reactnativejni)
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/cmake-utils/ReactNative-application.cmake:90]: This feels wrong @WoLewicki, I don't believe you need it.
Are you sure about this target?

**Review by cortinico — CHANGES_REQUESTED** [review]

File: packages/react-native/ReactAndroid/cmake-utils/ReactNative-application.cmake (hunk 1 above)
Comment (WoLewicki) [reply packages/react-native/ReactAndroid/cmake-utils/ReactNative-application.cmake:90]: It is needed in `TextLayoutManager` for `facebook::react::NativeArray::consume()`: https://github.com/facebook/react-native/blob/c4a6bbc8fd8992059abdcfae2bc483bd29f31e79/packages/react-native/ReactCommon/react/renderer/textlayoutmanager/platform/android/react/renderer/textlayoutmanager/TextLayoutManager.cpp#L193

image

**Review by cortinico — APPROVED** [review]

**Comment by cortinico** [comment]
@WoLewicki can you open a pick request for this one?

**Comment by WoLewicki** [comment]
Done

**Pull request closed without merging**